src/v2i-hub/PedestrianPlugin/src/include/FLIRWebSockAsyncClnSession.hpp

```diff
@@ -0,0 +1,93 @@
+#include <boost/beast/core.hpp>
+#include <boost/beast/websocket.hpp>
+#include <boost/asio/strand.hpp>
+#include <boost/bind.hpp>
+
+#include <cstdlib>
+#include <functional>
+#include <iostream>
+#include <memory>
+#include <string>
+
+namespace beast = boost::beast;         // from <boost/beast.hpp>
+namespace http = beast::http;           // from <boost/beast/http.hpp>
+namespace websocket = beast::websocket; // from <boost/beast/websocket.hpp>
+namespace net = boost::asio;            // from <boost/asio.hpp>
+using tcp = boost::asio::ip::tcp;       // from <boost/asio/ip/tcp.hpp>
+
+//------------------------------------------------------------------------------
+
+namespace PedestrianPlugin
+{
+    // Sends a WebSocket message and prints the response
+    class FLIRWebSockAsyncClnSession : public std::enable_shared_from_this<FLIRWebSockAsyncClnSession>
+    {
+        tcp::resolver resolver_;
+        websocket::stream<beast::tcp_stream> ws_;
+        beast::flat_buffer buffer_;
+        std::string host_;
+        std::string pedPresenceTrackingReq = std::string("{\"messageType\":\"Subscription\", \"subscription\":{ \"type\":\"Data\", \"action\":\"Subscribe\", \"inclusions\":[{\"type\":\"PedestrianPresenceTracking\"}]}}");
+        float cameraRotation_;
+        // std::string urlExtension_;
+        std::string psmxml = "";
+        std::mutex _psmLock;
+        int msgCount = 0;
+    public:
+
+    // Report a failure
+    void
+    fail(beast::error_code ec, char const* what)
+    {
+        std::cerr << what << ": " << ec.message() << "\n";
+    }
+        
+        // Resolver and socket require an io_context
+        explicit     
+        FLIRWebSockAsyncClnSession(net::io_context& ioc)
+            : resolver_(net::make_strand(ioc))
+            , ws_(net::make_strand(ioc))
+        {
+
+        };
+
+        // Start the asynchronous operation
+        void
+        run(
+            char const* host,
+            char const* port,
+            float cameraRotation);
+        
+
+        void
+        on_resolve(
+            beast::error_code ec,
+            tcp::resolver::results_type results);
+        
+
+        void
+        on_connect(beast::error_code ec, tcp::resolver::results_type::endpoint_type ep);
+                      
+        void
+        on_handshake(beast::error_code ec);
+
+        void
+        on_write(
+            beast::error_code ec,
+            std::size_t bytes_transferred);
+    
+
+        void
+        on_read(
+            beast::error_code ec,
```

Comment:
Can we add notes to explain these methods?

Reply:
@abey-yoseph  We try to leave java-doc style comments in the header file method declarations. If you need an example of java doc style comments check out header files in my most recent CARMA-Streets PR or online.

Reply:
Done